tests/security/389ds/tls_389ds_sssd_client.pm

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Implement & Integrate 389ds + sssd test case into openQA,
+#          This test module covers the sssd client tests
+#
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#88513, tc#1768672
+
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use lockapi;
+use Utils::Systemd qw(systemctl disable_and_stop_service);
+use version_utils 'is_tumbleweed';
+
+sub run {
+    select_console("root-console");
+
+    my $local_ip    = '10.0.2.102';
+    my $remote_ip   = '10.0.2.101';
+    my $local_name  = 'sssdclient';
+    my $remote_name = '389ds';
+    my $inst_ca_dir = '/etc/dirsrv/slapd-localhost';
+    my $tls_dir     = '/etc/openldap/certs';
+    my $ldap_user   = 'wilber';
+    my $uid         = '1003';
+
+    # Install 389-ds and sssd on client
+    zypper_call("in 389-ds sssd");
+
+    # Disable and stop the nscd daemon because it conflicts with sssd
+    disable_and_stop_service("nscd", ignore_failure => 1);
+
+    # Configure host names for C/S communication
+    assert_script_run("sed -i '/$local_ip.*/d' /etc/hosts");
+    assert_script_run("sed -i '/$remote_ip.*/d' /etc/hosts");
+    assert_script_run("echo '$local_ip $local_name.example.com $local_name' >> /etc/hosts");
+    assert_script_run("echo '$remote_ip $remote_name.example.com $remote_name' >> /etc/hosts");
```

Comment:
The same is done in the server side. Please create a perl routine for that.

Reply:
Thanks, let me do it